### PR TITLE
#167303083 create useForm hook

### PR DESCRIPTION
**Pull request by Eazybee**
#### What does this PR do?

- This PR creates useForm hook

#### Description of task to be completed?

- install @testing-library/react-hooks, validatorjs
- install react-test-renderer
- create a useForm.js in src/hooks
- crete useForm.spec.js in scr/hooks/test
- create useForm fixture in scr/fixtures/hooks

#### How should this be manually tested?

  - $ git clone https://github.com/andela/king-kong-ah-frontend.git
  - $ npm install
  - $ npm test

#### Any background context you want to provide?

- You would need npm to run this application

#### What are the relevant pivotal tracker stories?

- PT Story link - [#167303083](https://www.pivotaltracker.com/story/show/167303083)

#### Screenshots (if appropriate)
n/a

- Example
```
import React from 'react';
import useForm from '../../hooks/useForm';


const Form = () => {
  const login = (values) => {
    console.log('The form value object', values);
  };

  const inputRules = {
    username: 'alpha',
    age: 'numeric',
  };

  const {
    values, handleChange, handleSubmit,
  } = useForm(login, inputRules);

  const {
    username: { value: usernameValue, err: usernameErr },
    age: { value: ageValue, err: ageErr },
  } = values;

  return (
    <form onSubmit={handleSubmit}>
      <input
        type='text'
        value ={usernameValue}
        onChange={handleChange}
        name='username'
        required
      />
      {usernameErr && <p>{usernameErr}</p>}
      <br />
      <input
        type='text'
        value ={ageValue}
        onChange={handleChange}
        name='age'
      />
      {ageErr && <p>{ageErr}</p>}
      <br />
      <button type='submit'>Submit</button>
    </form>
  );
};

export default Form;

```

### CODE REVIEW

**Review by Sojisoyoye — APPROVED** [review]

**Review by akinyeleolat — APPROVED** [review]

**Review by frankly034 — APPROVED** [review]

File: src/fixtures/useForm.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    value: 'EazyBee',
+    name: 'username',
+  },
+};
+
+export const getEvent = (newObject) => {
+  return {
+    target: {
+      ...event.target,
+      ...newObject,
```
Comment (benfluleck) [inline src/fixtures/useForm.js]: What is newObject. Do you mean new Value ?

File: package.json [hunk 2]
```diff
@@ -91,6 +92,7 @@
     "lint-staged": "^9.0.0",
     "mini-css-extract-plugin": "^0.8.0",
     "react-styleguidist": "^9.1.11",
+    "react-test-renderer": "^16.8.6",
```
Comment (benfluleck) [inline package.json:95]: Why did you install this ?

Comment (benfluleck) [reply package.json:95]: Doesn't look like you are using it

Comment (Eazybee) [reply package.json:95]: Yes, when I installed `@testing-library/react-hooks` and tried to run the test, I got an error `cannot find react-test-renderer index.js  ../module...`.

I found the solution here https://github.com/testing-library/react-hooks-testing-library/issues/52 that I need to install this `react-test-rendere`

**Review by benfluleck — APPROVED** [review]

**Pull request merged**